src/components/assets/EvmNativeToken.vue

```diff
@@ -0,0 +1,138 @@
+              <div class="text--accent">
+                <token-balance :balance="String(bal)" :symbol="nativeTokenSymbol" />
+              </div>
+              <div class="text--label">
+                <span>{{ $n(balUsd) }} {{ $t('usd') }}</span>
+              </div>
+            </div>
+          </div>
+          <div class="column--asset-buttons column--buttons--native-token">
+            <router-link :to="buildTransferPageLink(nativeTokenSymbol)">
+              <button class="btn btn--sm">
+                {{ $t('assets.transfer') }}
+              </button>
+            </router-link>
+            <!-- Only SDN is able to bridge via cBridge at this moment -->
+            <a
+              v-if="nativeTokenSymbol === 'SDN'"
+              :href="cbridgeAppLink"
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+              <button class="btn btn--sm">
+                {{ $t('assets.bridge') }}
+              </button>
+            </a>
+            <button
+              v-if="isFaucet"
+              class="btn btn--sm"
+              @click="handleModalFaucet({ isOpen: true })"
+            >
+              {{ $t('assets.faucet') }}
+            </button>
+          </div>
+        </div>
+      </div>
+    </div>
+    <modal-faucet :is-modal-faucet="isModalFaucet" :handle-modal-faucet="handleModalFaucet" />
+  </div>
+</template>
+<script lang="ts">
+import { ethers } from 'ethers';
+import { $web3 } from 'src/boot/api';
+import { cbridgeAppLink } from 'src/c-bridge';
+import ModalFaucet from 'src/components/assets/modals/ModalFaucet.vue';
+import TokenBalance from 'src/components/common/TokenBalance.vue';
+import { faucetBalRequirement } from 'src/config/wallets';
+import { useAccount, useNetworkInfo, usePrice } from 'src/hooks';
+import { getTokenImage } from 'src/modules/token';
+import { buildTransferPageLink } from 'src/router/routes';
+import { useStore } from 'src/store';
+import { computed, defineComponent, ref, watchEffect } from 'vue';
+
+export default defineComponent({
+  components: { ModalFaucet, TokenBalance },
+  setup() {
+    const bal = ref<number>(0);
+    const balUsd = ref<number>(0);
+    const isShibuya = ref<boolean>(false);
+    const isRocstar = ref<boolean>(false);
+    const isFaucet = ref<boolean>(false);
+    const isModalFaucet = ref<boolean>(false);
+
+    const { currentNetworkName, nativeTokenSymbol } = useNetworkInfo();
+    const { currentAccount } = useAccount();
+    const { nativeTokenUsd } = usePrice();
+    const store = useStore();
+    const isH160 = computed<boolean>(() => store.getters['general/isH160Formatted']);
+    const isLoading = computed<boolean>(() => store.getters['general/isLoading']);
+
+    const nativeTokenImg = computed<string>(() =>
+      getTokenImage({ isNativeToken: true, symbol: nativeTokenSymbol.value })
+    );
+
+    const updateStates = async (nativeTokenUsd: number): Promise<void> => {
```

Comment:
I believe we should not have business or data logic in components. You don't need to do refactor now in this PR, but have in mind for the future components. I will try to remind myself also